plugins/dfu/dfu-target.c

```diff
@@ -472,6 +472,42 @@ dfu_target_status_to_error_msg (DfuStatus status)
 	return NULL;
 }
 
+gboolean
+dfu_target_poll_in_manifest (DfuTarget *target, GError **error)
+{
+	DfuTargetPrivate *priv = GET_PRIVATE (target);
+	DfuStatus status;
+
+	/* get the status */
+	if (!dfu_device_refresh (priv->device, error))
+		return FALSE;
+
+	/* wait for DFU_STATE_DFU_MANIFEST to not be set */
+	while (dfu_device_get_state (priv->device)
+					== DFU_STATE_DFU_MANIFEST_SYNC ||
+			dfu_device_get_state (priv->device)
+					== DFU_STATE_DFU_MANIFEST) {
+		g_debug ("waiting for DFU_STATE_DFU_MANIFEST to clear");
+		g_usleep ((dfu_device_get_download_timeout (
+					priv->device) + 1000) * 1000);
+		if (!dfu_device_refresh (priv->device, error))
+			return FALSE;
+	}
+
+	/* not in an error state */
+	if (dfu_device_get_state (priv->device) != DFU_STATE_DFU_ERROR)
+		return TRUE;
+
+	status = dfu_device_get_status (priv->device);
+	/* use a proper error description */
+	g_set_error_literal (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
```

Comment:
is this really the right description for the error?

Reply:
Checked the error code of FWUPD, maybe the "FWUPD_ERROR_INTERNAL" is more proper? The real error message returned by device is dfu_target_status_to_error_msg (status).